schema/query/predicate.go

```diff
@@ -545,3 +547,69 @@ func (e *Regex) Prepare(validator schema.Validator) error {
 func (e Regex) String() string {
 	return quoteField(e.Field) + ": {" + opRegex + ": " + valueString(e.Value) + "}"
 }
+
+// ElemMatch matches object values specified in an array.
+type ElemMatch struct {
+	Field string
+	Exps  []Expression
+}
+
+// Match implements Expression interface.
+func (e ElemMatch) Match(payload map[string]interface{}) bool {
+	value := getField(payload, e.Field)
+
+	var arr []interface{}
+	var ok bool
+	if arr, ok = value.([]interface{}); !ok {
+		return false
+	}
+
+	if e.Exps == nil || len(e.Exps) == 0 {
+		// nil or empty predicates will not match
+		return false
+	}
+	for _, val := range arr {
+		if v, ok := val.(map[string]interface{}); ok {
```

Comment:
It doesn't need to be an object for the query to match.

Examples of legal queries for an array of numbers for instance:
```js
{items: {$elemMatch: {$lt: 32}}}
{items: {$elemMatch: {$and:[{$lt: 32},{$gt: 5}]}}
```

In MongoDB, the latter one would be written more tidily using _implicit_ and `{items: {$elemMatch: {$lt: 32, $gt: 5}}}`, but that's does not currently work for comparison queries through rest-layer due to the way queries are parsed.

The docs states that `$elemMatch` is _not_ needed, if matching a single criteria, but it might be good to check weather or not it works. E.g.:

```js
{items: {$elemMatch: 6}}
```
If it works, that makes things easier.

Reply:
Yes, I guess `$elemMatch` can be more versatile. I will think a bit more, on how to make it work on array of values, instead of array of objects.

Reply:
The reason for mandatory `schema.Object`:
```
	if obj, ok = arr.Values.Validator.(*schema.Object); !ok {
		return fmt.Errorf("%s: array elements are not schema.Object", e.Field)
	}
	return prepareExpressions(e.Exps, obj.Schema)
```
Is that `prepareExpressions` needs `schema.Validator`, and not `schema.FieldValidator`.

Reply:
Ok, thanks for the explanation.

Let's keep it for objects only for now, and document it as a _current_ limitation.